packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,323 @@
+    });
+
+    const rolledSwaps = sortedSwaps?.reduce(
+      (acc, curr) => {
+        acc.totalAmountSwapped = acc.totalAmountSwapped.plus(curr.swapOutputAmount ?? 0);
+        if (curr.swapScheduledAt && curr.swapExecutedAt) {
+          acc.lastExecutedChunk = curr;
+          acc.totalChunksExecuted += 1;
+          acc.fees = acc.fees.concat(...curr.fees);
+        }
+        return acc;
+      },
+      {
+        totalAmountSwapped: new Prisma.Decimal(0),
+        totalChunksExecuted: 0,
+        currentChunk: sortedSwaps[0],
+        lastExecutedChunk: null as null | (typeof sortedSwaps)[number] | undefined,
+        isDcaSwap:
+          swapDepositChannel?.chunkIntervalBlocks && swapDepositChannel.chunkIntervalBlocks > 1,
+        fees: [] as SwapFee[],
+      },
+    );
+
+    const aggregateFees = rolledSwaps?.fees
+      .reduce((acc, curr) => {
+        const { type, asset, amount } = curr;
+
+        const index = acc.findIndex((fee) => fee.type === type && fee.asset === asset);
+        if (index !== -1) {
+          acc[index].amount = acc[index].amount.plus(amount);
+        } else acc.push(curr);
+
+        return acc;
+      }, [] as SwapFee[])
+      .concat(swapRequest?.fees ?? [])
+      .map((fee) => ({
+        type: fee.type,
+        ...getAssetAndChain(fee.asset),
+        amount: fee.amount.toFixed(),
+      }));
+
+    const showBoost = Boolean(
+      swapDepositChannel?.maxBoostFeeBps && swapDepositChannel.maxBoostFeeBps > 0,
+    );
+    const showCcm = Boolean(swapDepositChannel?.ccmGasBudget || swapDepositChannel?.ccmMessage);
+    const showRefund = swapDepositChannel?.fokRefundAddress;
+
+    const [
+      swapEgressFields,
+      refundEgressFields,
+      estimatedDurationSeconds,
+      srcChainRequiredBlockConfirmations,
+    ] = await Promise.all([
+      getEgressStatusFields(
+        swapEgress,
+        swapEgress?.broadcast,
+        ignoredEgresses,
+        'SWAP',
+        swapEgressTrackerTxRef,
+      ),
+      getEgressStatusFields(
+        refundEgress,
+        refundEgress?.broadcast,
+        ignoredEgresses,
+        'REFUND',
+        refundEgressTrackerTxRef,
+      ),
+      srcAsset && destAsset && estimateSwapDuration({ srcAsset, destAsset }),
+      internalSrcAsset && getRequiredBlockConfirmations(internalSrcAsset),
+    ]);
+
+    const response = {
+      state,
+      swapId: swapRequest?.nativeId.toString(),
+      ...(internalSrcAsset && getAssetAndChain(internalSrcAsset, 'src')),
+      ...(internalDestAsset && getAssetAndChain(internalDestAsset, 'dest')),
+      destAddress: readField(swapRequest, swapDepositChannel, failedSwap, 'destAddress'),
+      depositChannel: {
+        createdAt: swapDepositChannel?.createdAt.valueOf(),
+        brokerCommissionBps: swapDepositChannel?.brokerCommissionBps,
+        depositAddress: swapDepositChannel?.depositAddress,
+        expiryBlock: swapDepositChannel?.srcChainExpiryBlock?.toString(),
```

Comment:
would keep the `srcChain` prefix because we also have `retryDurationBlocks` which is a different kind of block 😅

Reply:
`srcChainExpiryBlock`